packages/redux/src/themes/reducer.ts

```diff
@@ -0,0 +1,82 @@
+import * as actionTypes from './actionTypes.js';
+import { type AnyAction, combineReducers, type Reducer } from 'redux';
+import type { ThemeState } from './types/index.js';
+
+export const INITIAL_STATE: ThemeState = {
+  error: {},
+  isLoading: {},
+  result: {},
+};
+
+const error = (state = INITIAL_STATE.error, action: AnyAction) => {
+  switch (action.type) {
+    case actionTypes.FETCH_THEME_SUCCESS:
+      return {
+        ...state,
+        [action.meta.code]: undefined,
+      };
+    case actionTypes.FETCH_THEME_FAILURE:
+      return {
+        ...state,
+        [action.meta.code]: action.payload.error,
+      };
+    default:
+      return state;
+  }
+};
+
+const isLoading = (state = INITIAL_STATE.isLoading, action: AnyAction) => {
+  switch (action.type) {
+    case actionTypes.FETCH_THEME_REQUEST:
+      return {
+        ...state,
+        [action.meta.code]: true,
+      };
+    case actionTypes.FETCH_THEME_SUCCESS:
+      return {
+        ...state,
+        [action.meta.code]: false,
+      };
+    case actionTypes.FETCH_THEME_FAILURE:
+      return {
+        ...state,
+        [action.meta.code]: undefined,
+      };
+    default:
+      return state;
+  }
+};
+
+const result = (state = INITIAL_STATE.result, action: AnyAction) => {
+  switch (action.type) {
+    case actionTypes.FETCH_THEME_SUCCESS:
```

Comment:
This should handle either `FETCH_THEME_FAILURE` or `FETCH_THEME_REQUEST` to clean up any previous results.

Reply:
Done